File: 1995/1/2-1.py
#!/usr/bin/env python

import sys

for line in sys.stdin:
    a,b = map(int, line.split())
    print "%s X %s =" % (a,b),
    sum = 0
    while a > 0:
        if a % 2:
            print b,
            sum += b
            if a > 1:
                print "+",
        a /= 2
        b *= 2
    print "= %s" % sum

File: 1995/1/2-1.rb
#!/usr/bin/env ruby

File.readlines('prob1.in').each do |line|
  a,b = line.split().collect { |n| n.to_i }
  print "#{a} * #{b} = "
  sum = 0
  while a > 0
    if a % 2 > 0
      print b
      sum += b
      print " + " if a > 1
    end
    a /= 2
    b *= 2
  end
  print " = #{sum}"
  puts
end
